api/src/main/java/org/openmrs/module/esaudereports/reporting/library/cohort/SaprAprCohort.java

```diff
@@ -0,0 +1,251 @@
+		
+		Concept START = Context.getConceptService().getConceptByUuid(Metadata._Concept.START);
+		
+		Concept TRANSFERED_FROM = Context.getConceptService().getConceptByUuid(Metadata._Concept.TRANSFERED_FROM);
+		
+		cd.setName("ALGUMA VEZ ESTEVE EM TRATAMENTO ARV - PERIODO FINAL");
+		cd.setDescription("Pacientes que alguma vez esteve em tratamento ARV (Iniciou ou veio transferido de outra us em TARV) até um determinado periodo final");
+		cd.setQuestion(Context.getConceptService().getConceptByUuid(Metadata._Concept.ART_MANAGEMENT));
+		cd.setEncounterTypeList(Arrays.asList(ADULTO_SEGUIMENTO, PEDIATRIA_SEGUIMENTO, FARMACIA));
+		cd.setTimeModifier(PatientSetService.TimeModifier.ANY);
+		cd.setOperator(SetComparator.IN);
+		cd.setValueList(Arrays.asList(START, TRANSFERED_FROM));
+		cd.addParameter(new Parameter("location", "Unidade Sanitaria", Location.class));
+		//		cd.addParameter(new Parameter("startDate", "Data Inicial", Date.class));
+		cd.addParameter(new Parameter("endDate", "ATE", Date.class));
+		
+		return cd;
+	}
+	
+	public CohortDefinition patientsInscribedOnARTProgram() {
+		SqlCohortDefinition cd = new SqlCohortDefinition();
+		cd.setName("PROGRAMA: PACIENTES INSCRITOS NO PROGRAMA TRATAMENTO ARV (TARV) - PERIODO FINAL");
+		cd.setDescription("Sao pacientes inscritos no programa de tratamento ARV até um determinado periodo final");
+		cd.addParameter(new Parameter("endDate", "Data Inicial", Date.class));
+		cd.addParameter(new Parameter("location", "US", Location.class));
+		cd.setQuery(CohortQueries.PATIENTS_INSCRIBED_ON_ART_PROGRAM);
+		
+		return cd;
+	}
+	
+	public CohortDefinition aRTStartUsingDateConcept() {
+		DateObsCohortDefinition cd = new DateObsCohortDefinition();
+		
+		EncounterType ADULTO_SEGUIMENTO = Context.getEncounterService().getEncounterType(
+		    Metadata._EncounterType.ADULTO_SEGUIMENTO_6);
+		
+		EncounterType PEDIATRIA_SEGUIMENTO = Context.getEncounterService().getEncounterType(
+		    Metadata._EncounterType.PEDIATRIA_SEGUIMENTO_9);
+		
+		EncounterType FARMACIA = Context.getEncounterService().getEncounterType(Metadata._EncounterType.FARMACIA_18);
+		
+		cd.setName("INICIO DE TARV USANDO O CONCEITO DE DATA - PERIODO FINAL");
+		cd.setDescription("São pacientes que iniciaram TARV registado no conceito 'Data de Inicio de TARV' na ficha de seguimento");
+		cd.setQuestion(Context.getConceptService().getConceptByUuid(Metadata._Concept.ART_START_DATE));
+		cd.setEncounterTypeList(Arrays.asList(ADULTO_SEGUIMENTO, PEDIATRIA_SEGUIMENTO, FARMACIA));
+		cd.setTimeModifier(PatientSetService.TimeModifier.ANY);
+		cd.addParameter(new Parameter("location", "Unidade Sanitaria", Location.class));
+		cd.setOperator1(RangeComparator.LESS_EQUAL);
+		cd.addParameter(new Parameter("endDate", "Data Final", Date.class));
+		
+		return cd;
+	}
+	
+	public CohortDefinition haveBeenInART_Pharmacy() {
+		EncounterCohortDefinition cd = new EncounterCohortDefinition();
```

Comment:
can https://github.com/esaude/esaude-reports/blob/master/api/src/main/java/org/openmrs/module/esaudereports/reporting/library/cohort/CommonCohortLibrary.java#L103 be utilized instead of code duplication